--- a/netlink/Makefile	2024-05-10 15:07:23.479127795 +0000
+++ b/netlink/Makefile	2024-05-10 22:37:44.207522227 +0000
@@ -1,3 +1,4 @@
+GRE_ENABLED := $(strip $(if $(filter $(gre), y), 1 , 0))
 CAPWAP_ENABLED := $(strip $(if $(filter $(capwapmgr), y), 1 , 0))
 IPSEC_ENABLED := $(strip $(if $(filter $(ipsecmgr), y), 1 , 0))
 DTLS_ENABLED := $(strip $(if $(filter $(dtlsmgr), y), 1 , 0))
@@ -11,7 +12,7 @@
 ccflags-y += -DCONFIG_NSS_NLIPV4=1
 ccflags-y += -DCONFIG_NSS_NLIPV6=1
 ccflags-y += -DCONFIG_NSS_NLOAM=1
-ccflags-y += -DCONFIG_NSS_NLGRE_REDIR_FAMILY=1
+ccflags-y += -DCONFIG_NSS_NLGRE_REDIR_FAMILY=${GRE_ENABLED}
 ccflags-y += -DCONFIG_NSS_NLETHRX=1
 ccflags-y += -DCONFIG_NSS_NLDYNAMIC_INTERFACE=1
 ccflags-y += -DCONFIG_NSS_NLN2H=1
@@ -29,11 +30,13 @@
 ccflags-y += -DCONFIG_NSS_NLDTLS=${DTLS_ENABLED}
 
 qca-nss-netlink-objs := nss_nl.o
+ifneq (,$(filter $(gre), y))
 qca-nss-netlink-objs += nss_nlgre_redir_family.o
 qca-nss-netlink-objs += nss_nlgre_redir_cmd.o
 qca-nss-netlink-objs += nss_nlgre_redir_cmn.o
 qca-nss-netlink-objs += nss_nlgre_redir.o
 qca-nss-netlink-objs += nss_nlgre_redir_lag.o
+endif
 qca-nss-netlink-objs += nss_nlipv4.o
 qca-nss-netlink-objs += nss_nlipv6.o
 qca-nss-netlink-objs += nss_nloam.o
@@ -44,11 +47,13 @@
 qca-nss-netlink-objs += nss_nlipv6_reasm.o
 qca-nss-netlink-objs += nss_nlwifili.o
 qca-nss-netlink-objs += nss_nllso_rx.o
+ifneq (,$(filter $(map_t), y))
 qca-nss-netlink-objs += nss_nlmap_t.o
+endif
 qca-nss-netlink-objs += nss_nlpppoe.o
 qca-nss-netlink-objs += nss_nll2tpv2.o
 qca-nss-netlink-objs += nss_nlpptp.o
-qca-nss-netlink-objs += nss_nlqrfs.o
+#qca-nss-netlink-objs += nss_nlqrfs.o
 
 ifneq (,$(filter $(capwapmgr), y))
 qca-nss-netlink-objs += nss_nlcapwap.o
