--- a/nss_core.c
+++ b/nss_core.c
@@ -1117,7 +1117,7 @@ static inline bool nss_core_handle_nr_fr
 		 * ingress shaped flag set.
 		 */
 		if (unlikely(desc->bit_flags & N2H_BIT_FLAG_INGRESS_SHAPED)) {
-			nbuf->tc_verd = SET_TC_NCLS_NSS(nbuf->tc_verd);
+			nbuf->tc_verd_qca_nss = SET_TC_NCLS_NSS(nbuf->tc_verd_qca_nss);
 		}
 #endif
 		goto pull;
@@ -1157,7 +1157,7 @@ static inline bool nss_core_handle_nr_fr
 		 * ingress shaped flag set.
 		 */
 		if (unlikely(desc->bit_flags & N2H_BIT_FLAG_INGRESS_SHAPED)) {
-			nbuf->tc_verd = SET_TC_NCLS_NSS(nbuf->tc_verd);
+			nbuf->tc_verd_qca_nss = SET_TC_NCLS_NSS(nbuf->tc_verd_qca_nss);
 		}
 #endif
 
@@ -1272,7 +1272,7 @@ static inline bool nss_core_handle_linea
 		 * ingress shaped flag set.
 		 */
 		if (unlikely(desc->bit_flags & N2H_BIT_FLAG_INGRESS_SHAPED)) {
-			nbuf->tc_verd = SET_TC_NCLS_NSS(nbuf->tc_verd);
+			nbuf->tc_verd_qca_nss = SET_TC_NCLS_NSS(nbuf->tc_verd_qca_nss);
 		}
 #endif
 
@@ -1330,7 +1330,7 @@ static inline bool nss_core_handle_linea
 		 * ingress shaped flag set.
 		 */
 		if (unlikely(desc->bit_flags & N2H_BIT_FLAG_INGRESS_SHAPED)) {
-			nbuf->tc_verd = SET_TC_NCLS_NSS(nbuf->tc_verd);
+			nbuf->tc_verd_qca_nss = SET_TC_NCLS_NSS(nbuf->tc_verd_qca_nss);
 		}
 #endif
 
