--- a/ecm_interface.c	2024-05-09 09:05:49.741268323 +0000
+++ b/ecm_interface.c	2024-05-09 09:05:42.773255261 +0000
@@ -782,6 +782,27 @@
 }
 EXPORT_SYMBOL(ecm_interface_mac_addr_get);
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(5, 11, 0))
+static void ecm_interface_br_dev_update_stats(struct net_device *dev,
+                                             struct rtnl_link_stats64 *nlstats)
+{
+       struct pcpu_sw_netstats *tstats;
+
+       /* Is this a bridge? */
+       if (!(dev->priv_flags & IFF_EBRIDGE))
+               return;
+
+       tstats = this_cpu_ptr(dev->tstats);
+
+       u64_stats_update_begin(&tstats->syncp);
+       tstats->rx_packets += nlstats->rx_packets;
+       tstats->rx_bytes += nlstats->rx_bytes;
+       tstats->tx_packets += nlstats->tx_packets;
+       tstats->tx_bytes += nlstats->tx_bytes;
+       u64_stats_update_end(&tstats->syncp);
+}
+#endif
+
 #ifdef ECM_IPV6_ENABLE
 /*
  * ecm_interface_mac_addr_get_ipv6_no_route()
@@ -6753,7 +6774,11 @@
 			stats.rx_bytes = rx_bytes;
 			stats.tx_packets = tx_packets;
 			stats.tx_bytes = tx_bytes;
-			br_dev_update_stats(dev, &stats);
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(5, 11, 0))
+                        ecm_interface_br_dev_update_stats(dev, &stats);
+#else
+                        br_dev_update_stats(dev, &stats);
+#endif
 			dev_put(dev);
 			continue;
 
