--- a/Makefile	2024-05-08 06:28:10.653788053 +0000
+++ b/Makefile	2024-05-08 06:28:33.593831878 +0000
@@ -333,7 +333,6 @@
 # the Parental Controls subsystem classifier in ECM. Currently disabled until
 # customers require it / if they need to integrate their Parental Controls with it.
 # #############################################################################
-ECM_CLASSIFIER_PCC_ENABLE=y
 ecm-$(ECM_CLASSIFIER_PCC_ENABLE) += ecm_classifier_pcc.o
 ccflags-$(ECM_CLASSIFIER_PCC_ENABLE) += -DECM_CLASSIFIER_PCC_ENABLE
 
